src/java.base/share/classes/java/security/KeyStoreSpi.java

```diff
@@ -316,6 +319,8 @@ public abstract void engineStore(OutputStream stream, char[] password)
      *          algorithm could not be found
      * @throws    CertificateException if any of the certificates included in
      *          the keystore data could not be stored
+     * @throws UnsupportedOperationException if the implementation
+     *          has not overridden this method
```

Comment:
I think we should keep this more general and just say "If the implementation does not support this operation". There may be legitimate cases in which an implementation may override this method but throw UnsupportedOperationException if it doesn't support particular aspects of the LoadStoreParameter.

Reply:
Good point. Fixed as suggested.